### DISCUSSION TITLE
Add support for vitest@0.34.0 `--experimentalVmThreads`

### DISCUSSION THREAD

**Discussion by cmolina** [mswjs/msw#1706] [Ideas]
Hi, I recently read about [this new vitest flag](https://vitest.dev/config/#experimentalvmthreads) that should allow to run tests faster by using Node's VM context.

However, when I tried to run my tests, they all failed with this error, just before trying to import `msw`'s `setupServer`:

```output
Error: Dynamic require of "util" is not supported
 ❯ ../../node_modules/@mswjs/interceptors/lib/node/chunk-6CRMKMDL.mjs:6:9
 ❯ ../../node_modules/@mswjs/interceptors/lib/node/chunk-RFVEKRYP.mjs:6:67
 ❯ ../../test/javascript/setup-tests.ts:1:31
```

**Comment by kettanaito** [comment]
Hey, @cmolina. Thanks for proposing this, I was not aware of this new flag being added to Vitest. 

Since it implies threads, it likely means Vitest will split tests across different Node.js threads, running them in parallel. Every thread implies a new Node.js process but given it's Vitest's responsibility to manage those, I'd expect Vitest to still call its setup hooks, like `beforeAll` and `afterAll`, which will enable MSW even in those spawned processes. Functionality-wise, I'd expect MSW to work with this out of the box. 

It looks like the error you shared is related to some `require()` statements. That's odd. Do you see how the code looks like at the printed stack trace (`node_modules/@mswjs/interceptors/lib/node/chunk-6CRMKMDL.mjs:6:9`)? I wonder why _something_ thinks there's a dynamic require—we don't do those in our source code (may be a compilation issue though). If you're using any kind of code pre-processing for your tests, try turning that off and see if the error changes/goes away. 

As with any issue, I would highly appreciate a reproduction repository for this!
